### ISSUE TITLE
Allow `_set` in `on_conflict`

### ISSUE THREAD

**Issue by webdeb** [hasura/graphql-engine#5425]
## Use case

A user can buy credits which are stored in a `credits` table, which has an unique `user_id` field
I would like to insert the credits entry for a user and use `on_conflict` to increment the amount value if there is already a credits row for this user.

## Current behaviour

`on_conflict` only supports the column to be updated overridden instead of an operation like `_set`

## Proposal

Allow `_set` in `on_conflict` to have a flexible way to update the data.

**Comment by yasinarik** [comment]
I agree with @webdeb . His workaround is a way of doing that. However, there are some disadvantages of doing it using the current features.
First of all, as far as I know, the `increment` `_inc` operation is **atomic**. 
Secondly, when I want to update a table with 10+ fields to be incremented, I have to pass all of them as a parameter one by one. In that case, using an array-object with such an operation: `on_conflict` --> `_inc` will be really useful and efficient.

I hope that I have explained it adequately. If not, please let me know and will try to illustrate better.

### For example: 
#### (Just think that all that `10`s below are replaced with the corresponding variable names.)
```graphql
  update_userstats(where: {user_id: {_eq: $user_id}}, _inc: {count_gold_bars: 10, count_gold_coins: 10, count_ruby: 10, score: 10, successful_games: 10, successful_rounds: 10, total_games: 10, total_rounds: 10}) {
    affected_rows
  }
```
